### PR TITLE
dnsdist: Add metrics for health-check failures

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Closes https://github.com/PowerDNS/pdns/issues/12846.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdistdist/dnsdist-healthchecks.cc [hunk 1]
```diff
@@ -151,6 +158,7 @@ class HealthCheckQuerySender : public TCPQuerySender
 
   void notifyIOError(InternalQueryState&& query, const struct timeval& now) override
   {
+    ++d_data->d_ds->d_healthCheckMetrics.d_networkErrors;
```
Comment (omoerbeek) [inline pdns/dnsdistdist/dnsdist-healthchecks.cc:161]: Several cases of `d_networkErrors` are lumped together. Does it make sense to separate those?

File: pdns/dnsdistdist/dnsdist-healthchecks.cc [hunk 2]
```diff
@@ -471,6 +483,7 @@ void handleQueuedHealthChecks(FDMultiplexer& mplexer, bool initial)
           infolog("Timeout while waiting for the health check response (ID %d) from backend %s", data->d_queryID, data->d_ds->getNameWithAddr());
         }
 
+        ++data->d_ds->d_healthCheckMetrics.d_timeOuts;
         data->d_ds->submitHealthCheckResult(initial, false);
       }
       catch (const std::exception& e) {
```
Comment (omoerbeek) [inline pdns/dnsdistdist/dnsdist-healthchecks.cc:489]: No increment of any counter here, some a few other cases in `handleQueuedHealthChecks()`. I'm not  sure if these cases are already handled by the callback functions or not. At least a comment would be nice.

**Review by omoerbeek — COMMENTED** [review]
I have two questions.

File: pdns/dnsdistdist/dnsdist-healthchecks.cc (hunk 1 above)
Comment (rgacogne) [reply pdns/dnsdistdist/dnsdist-healthchecks.cc:161]: I don't think so, as my current plan is to get a better understanding of the generic kind of failures. If we want more details we can turn the "verbose healthchecks" option to get detailed errors in the logs.

File: pdns/dnsdistdist/dnsdist-healthchecks.cc (hunk 2 above)
Comment (rgacogne) [reply pdns/dnsdistdist/dnsdist-healthchecks.cc:493]: I added a couple comments

**Review by omoerbeek — APPROVED** [review]

**Pull request merged**